### DISCUSSION TITLE
Trying to understand how "next-version" in GitVersion.yml works

### DISCUSSION THREAD

**Discussion by koonfoon** [GitTools/GitVersion#2641] [Q&A] [answered]
I am confused about how "next-version" property works. I read the official documentation, it just a two line "server as base version". I assume it will bump to the version I set on the next commit. But apparently it does not work like that. Hope some one can explain it further. Thank you.

**Comment by asbjornu** [comment, marked as answer]
[I have answered this on Stack Overflow](https://stackoverflow.com/questions/66950938/trying-to-understand-how-next-version-property-in-gitversion-yml-works):

> [`next-version` is described in the documentation](https://gitversion.net/docs/reference/configuration#next-version) as:
>
> Allows you to bump the next version explicitly, useful for bumping `main` or a feature with breaking changes a major increment.
>
> You can think of `next-version` as a "future" `git tag`. The value of `next-version` should be equal to the value of the `git tag` you are going to add to your next release commit (a merge commit from `develop` to `main`, for instance).
>
> As GitVersion is [configured to `increment`](https://gitversion.net/docs/reference/configuration#increment) either `Major`, `Minor` or `Patch`, for some branches you may want to exercise more control over which version part is incremented. Say GitVersion is configured to `increment: Patch`, but you know that there are features in `develop` that warrant a `Minor` or perhaps even `Major` increment. You can then use `next-version` to explicitly control the increment.
